src/utils/cpu.js

```diff
@@ -0,0 +1,201 @@
+import calculateWinner from "../utils/calculateWinner";
+import canPutStone from "./canPutStone";
+import getLowestEmptyYIndex from "./getLowestEmptyYIndex";
+
+const GameState = {
+  game: "GAME",
+  player_win: "PLAYER_WIN",
+  cpu_win: "CPU_WIN",
+  draw: "DRAW",
+};
+
+/**
+ * CPUのクラス
+ */
+class Cpu {
+  /**
+   * Cpuクラスのコンストラクターの説明
+   * @param {string[][]} board - 盤面を表す二次元配列
+   * @param {number} victoryCondition - 勝利条件
+   * @param {string} cpuName - boardで使用しているcpuの名前（Player2など）
+   * @param {string} playerName - boardで使用しているplayerの名前（Player1など）
+   */
+  constructor(board, victoryCondition, cpuName, playerName) {
+    this.board = board;
+    this.victoryCondition = victoryCondition;
+    this.cpuName = cpuName;
+    this.playerName = playerName;
+    this.myTurn = false; // false: CPU, true: Player
+    this.state = GameState.game;
+  }
+
+  /**
+   * ターンを交代する
+   */
+  changeTurn() {
+    this.myTurn = !this.myTurn;
+  }
+
+  /**
+   * 座標に対する操作を取り消す
+   * @param {number} x - 操作を取り消す座標x
+   * @param {number} y - 操作を取り消す座標y
+   */
+  undoValue(x, y) {
+    this.board[x][y] = null;
+    this.changeTurn();
+  }
+
+  /**
+   * 指定の座標に石を置く
+   * @param {number} x - 石を置く座標x
+   * @param {number} y - 石を置く座標y
+   */
+  putStone(x, y) {
+    this.myTurn ? (this.board[x][y] = this.playerName) : (this.board[x][y] = this.cpuName);
+    this.checkState(x, y);
+    this.changeTurn();
+  }
+
+  /**
+   * ゲームの状態を更新する
+   * @param {number} x - 直前に石を置いた座標x
+   * @param {number} y - 直前に石を置いた座標y
+   */
+  checkState(x, y) {
+    let winner = calculateWinner(this.board, this.victoryCondition, x, y);
```

Comment:
```suggestion
    const winner = calculateWinner(this.board, this.victoryCondition, x, y);
```
再代入しない値に関しては、なるべく`const`のほうがいいです